pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -55,22 +57,37 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		}
 	}
 
-	if host, port, err := net.SplitHostPort(obj.HealthzBindAddress); err == nil {
+	defaultBindAddress := net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+	if obj.HealthzBindAddress == nil {
+		obj.HealthzBindAddress = &defaultBindAddress
+	}
+	if host, port, err := net.SplitHostPort(*obj.HealthzBindAddress); err == nil {
 		if len(host) == 0 {
 			host = "0.0.0.0"
 		}
-		obj.HealthzBindAddress = net.JoinHostPort(host, port)
-	} else {
-		obj.HealthzBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+		hostPort := net.JoinHostPort(host, port)
+		obj.HealthzBindAddress = &hostPort
 	}
 
-	if host, port, err := net.SplitHostPort(obj.MetricsBindAddress); err == nil {
+	if obj.MetricsBindAddress == nil {
+		obj.MetricsBindAddress = &defaultBindAddress
+	}
+	if host, port, err := net.SplitHostPort(*obj.MetricsBindAddress); err == nil {
 		if len(host) == 0 {
 			host = "0.0.0.0"
 		}
-		obj.MetricsBindAddress = net.JoinHostPort(host, port)
-	} else {
-		obj.MetricsBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+		hostPort := net.JoinHostPort(host, port)
+		obj.MetricsBindAddress = &hostPort
+	}
+
+	if obj.DisablePreemption == nil {
+		disablePreemption := false
+		obj.DisablePreemption = &disablePreemption
+	}
+
+	if obj.PercentageOfNodesToScore == nil {
+		percentageOfNodesToScore := int32(50)
```

Comment:
please move the const in [api/types.go](https://github.com/kubernetes/kubernetes/blob/b8b7c376edb62ce8be42cef3728112c5c71d9fa4/pkg/scheduler/api/types.go#L29) to apis/config/types.go and use it here.

Reply:
done